modules/features/account/src/main/java/au/com/shiftyjelly/pocketcasts/account/viewmodel/ProfileUpgradeBannerViewModel.kt

```diff
@@ -105,4 +133,50 @@ class ProfileUpgradeBannerViewModel @Inject constructor(
             settings.setLastSelectedSubscriptionTier(upgradeFeatureCard.subscriptionTier)
         }
     }
+
+    private fun getDefaultSubscription(
+        filteredSubscriptions: List<Subscription>,
+        cachedTier: SubscriptionTier?,
+        cachedSubscriptionStatus: SubscriptionStatus?,
+    ) = subscriptionManager.getDefaultSubscription(
+        subscriptions = filteredSubscriptions,
+        tier = if (cachedTier == SubscriptionTier.PATRON) {
+            Subscription.SubscriptionTier.PATRON
+        } else {
+            Subscription.SubscriptionTier.PLUS
+        },
```

Comment:
Instead of using an if-else I would use `when` here covering all the possible cases. 

``` kotlin
tier = when (cachedTier) {
    SubscriptionTier.PATRON -> Subscription.SubscriptionTier.PATRON
    
    null,
    SubscriptionTier.NONE,
    SubscriptionTier.PLUS -> Subscription.SubscriptionTier.PLUS
},
```

I like how it is more explicit about the cases it is covering (i.e., what we're doing for `null` and `NONE`), and if we ever add a new tier we'll be forced to explicitly handle it here instead of just defaulting to Plus. Not a blocker though.

Reply:
The account details upgrade view is shown only when users have a free or expiring account. But you made a valid point that it's good to handle all cases explicitly from the new tier perspective. I updated it in a613822.